frontend/src/components/headers/DocumentHeader.tsx

```diff
@@ -23,13 +26,29 @@ import DownloadMenu from "../common/DownloadMenu";
 import ShareButton from "../common/ShareButton";
 import ThemeButton from "../common/ThemeButton";
 import UserPresenceList from "./UserPresenceList";
+import { FormContainer, TextFieldElement } from "react-hook-form-mui";
+import { selectDocument, setDocumentData } from "../../store/documentSlice";
+import { useUpdateDocumentTitleMutation } from "../../hooks/api/workspaceDocument";
+import { UpdateDocumentRequest } from "../../hooks/api/types/document";
 
 function DocumentHeader() {
 	const dispatch = useDispatch();
 	const navigate = useNavigate();
 	const editorState = useSelector(selectEditor);
 	const workspaceState = useSelector(selectWorkspace);
+	const documentStore = useSelector(selectDocument);
 	const { presenceList } = useUserPresence(editorState.doc);
+	const [focused, setFocused] = useState(false);
+	const { mutateAsync: updateDocumentTitle } = useUpdateDocumentTitleMutation(
+		workspaceState.data?.id || "",
+		documentStore.data?.id || ""
+	);
+
+	const isEditingDisabled = editorState.shareRole === "READ";
```

Comment:
It should be `Boolean(editorState.shareRole)`.
The document title cannot be changed in share mode.